nearcore/src/config.rs

```diff
@@ -1039,87 +1056,191 @@ pub fn create_testnet_configs_from_seeds(
-                first_node_addr.to_string()
-            } else {
-                tcp::ListenerAddr::reserve_for_test().to_string()
-            };
-            config.set_rpc_addr(tcp::ListenerAddr::reserve_for_test());
-            config.network.boot_nodes = if i == 0 {
-                "".to_string()
-            } else {
-                format!("{}@{}", network_signers[0].public_key, first_node_addr)
-            };
-            config.network.skip_sync_wait = num_validator_seats == 1;
-        }
-        config.archive = archive;
-        config.tracked_shards.clone_from(&tracked_shards);
-        config.consensus.min_num_peers =
-            std::cmp::min(num_validator_seats as usize - 1, config.consensus.min_num_peers);
-        configs.push(config);
+
+    // We assign the seeds to the nodes in the following order:
+    // 1. Validators (num_validators)
+    // 2. Non-validator archival nodes (num_non_validators_archival)
+    // 3. Non-validator RPC nodes (num_non_validators_rpc)
+    // 4. Non-validator nodes (num_non_validators)
+
+    // We use the first validator node as the boot node.
+    assert!(num_validators > 0, "No validators were added");
+    let boot_node_addr = tcp::ListenerAddr::reserve_for_test();
+    for i in 0..num_validators {
+        configs.push(create_localnet_config(
+            num_shards,
+            num_validators,
+            &tracked_shards,
+            &network_signers,
+            &boot_node_addr,
+            LocalnetNodeParams {
+                is_boot: (i == 0),
+                is_validator: true,
+                is_archival: false,
+                is_rpc: false,
+            },
+        ));
+    }
+    for _ in 0..num_non_validators_archival {
+        configs.push(create_localnet_config(
+            num_shards,
+            num_validators,
+            &tracked_shards,
+            &network_signers,
+            &boot_node_addr,
+            LocalnetNodeParams {
+                is_boot: false,
+                is_validator: false,
+                is_archival: true,
+                is_rpc: false,
+            },
+        ));
+    }
+    for _ in 0..num_non_validators_rpc {
+        configs.push(create_localnet_config(
+            num_shards,
+            num_validators,
+            &tracked_shards,
+            &network_signers,
+            &boot_node_addr,
+            LocalnetNodeParams {
+                is_boot: false,
+                is_validator: false,
+                is_archival: false,
+                is_rpc: true,
+            },
+        ));
+    }
+    for _ in 0..num_non_validators {
+        configs.push(create_localnet_config(
+            num_shards,
+            num_validators,
+            &tracked_shards,
+            &network_signers,
+            &boot_node_addr,
+            LocalnetNodeParams {
+                is_boot: false,
+                is_validator: false,
+                is_archival: false,
+                is_rpc: false,
```

Comment:
Sanity check - Is this the current behaviour? I'm assuming you want to make this PR a no-op for the existing nayduck tests.

Reply:
yes, actually the tests just create plain non-validators and then configure them within the test body. I plan to change that in follow-up PRs.